README.md

```diff
@@ -1,19 +1,127 @@
 [Device42](http://www.device42.com/) is a Continuous Discovery software for your IT Infrastructure. It helps you automatically maintain an up-to-date inventory of your physical, virtual, and cloud servers and containers, network components, software/services/applications, and their inter-relationships and inter-dependencies.
 
-This repository has 2 different scripts that you can use with ansible.
-1. `d42_ansible_inventory_hostfile.py` can be used to create a static inventory file for ansible. You can group hosts by tags, customers, building or service level from Device42 data.
-2. `d42_ansible_dynamic_inventory.py` can be used to dynamically by ansible to get hosts from Device42 based on certain filters.
+This Ansible Collection contains the inventory and lookup plugins for Device42. 
+1. Inventory (device42.d42.d42) - can be used to dynamically generate Ansible inventory from Device42 devices.
+2. Lookup (device42.d42.d42)- This lookup plugin allows a playbook to fetch passwords or ip addresses for hosts from Device42.
+3. Lookup (device42.d42.d42_prompt) - This lookup plugin allows a playbook to fetch passwords or ip addresses for hosts from Device42 with prompt. 
 
+In the `contrib` directory you will find the legacy inventory scripts. Please favor the plugin over the legacy scripts:
+1. `contrib\inventory\d42_ansible_inventory_hostfile.py` can be used to create a static inventory file for ansible. You can group hosts by tags, customers, building or service level from Device42 data.
+2. `contrib\inventory\d42_ansible_dynamic_inventory.py` can be used to dynamically by ansible to get hosts from Device42 based on certain filters.
+
+-----------------------------
 ## Assumptions
+- This script works with Device42 12.0.0 and above
+### Requirements
+- ansible 2.9+
+- python 3.6.x+
+- requests (you can install it with pip install requests or apt-get install python-requests)
+- Ansible must have an available connection to your Device42 instance in order to collect devices for inventory
+
+### Installation
+#### Ansible Tower:
+* Clone the repo on the host of your Anisble server
+```bash
+git clone git@github.com:device42/ansible_device42.git
+```
+* Navigate to the cloned directory
+```bash
+cd path/to/directory
+```
+* Build the Ansible collection
+```bash
+ansible-galaxy collection build
+```
+* Install the Ansible collection 
+```bash
+ansible-galaxy collection install device42-d42-2.0.0.tar.gz -p ../.ansible/collections
+```
+
+* Set the following environmental variables (optional)
```

Comment:
If you would like to include the [Custom Credential](https://docs.ansible.com/ansible-tower/latest/html/userguide/credential_types.html) configuration for Tower in the README here they are. 
Input Configuration:
```
fields:
  - type: string
    id: url
    label: URL
  - type: string
    id: username
    label: Username
  - type: string
    id: password
    label: Password
    secret: true
  - type: boolean
    id: skip_ssl
    label: Skip SSL Check
required:
  - url
  - username
  - password
```
Injector Configuration
```
env:
  - D42_URL: "{{ url }}"
  - D42_USER: "{{ username }}"
  - D42_PWD: "{{ password }}"
  - D42_SKIP_SSL_CHECK: "{{ skip_ssl }}"
```